clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -563,10 +615,15 @@ public synchronized void removeReporter(MetricsReporter reporter) {
         }
     }
 
-    synchronized void registerMetric(KafkaMetric metric) {
+    synchronized void registerMetric(KafkaMetric metric, boolean raiseIfMetricExists) {
```

Comment:
Would it be useful to know if the metric is created vs. already existed?

Reply:
Yeah currently, metricOrElseCreate hides that fact. I can add a log line here is needed. Also, I guess the addMetric method(s) already can be used to answer the created v/s already existed question.

Reply:
@jolshan I added a log line for the case where there's no exception to be thrown.

Reply:
I am wonder if this change is really necessary. Is it?

Reply:
Hey @vamossagar12 @jolshan @dajac , since `registerMetric` is a package private function I'm thinking about modifying its signature to return the already existed metric (null if it does not exist, not null means no changes to the underlying registry since there's already a metric), hence indicating if the register-metric succeeded or not. And let the caller (`addMetric` or `metricOrElseCreate`) to react accordingly: e.g. the former would still throw exception if it gets a non-null return value, while the latter just returns the metric. WDYT?

Reply:
@guozhangwang i think yeah that could be done. I have made the changes.

Reply:
`registerMetric` is also called from the `Sensor` class so we have to check if that works there as well.

Reply:
There were a couple of test failures because of this usage. I updated the code to throw an `IllegalArgumentException` for backward compatibility reasons.